time.go

```diff
@@ -227,6 +228,15 @@ func (t DateTime) MarshalBSONValue() (bsontype.Type, []byte, error) {
 // assumed to be valid. UnmarshalBSONValue must copy the BSON value bytes if it
 // wishes to retain the data after returning.
 func (t *DateTime) UnmarshalBSONValue(tpe bsontype.Type, data []byte) error {
```

Comment:
Maybe we should wrap https://github.com/mongodb/mongo-go-driver/blob/0cc7c8bc77bcb21feae730eb4e6de31609d3545b/bson/primitive/primitive.go#L52 instead of reinvent the Unmarshal.

Reply:
This might be a good idea but it will break existing code. I suggest we fix it first so it doesn't panic on nulls then change the underlying type (in another PR). This way users can decide when to upgrade  and refactor their code.

Reply:
Don't change the underlying type. use the bson type to Unmarshal and then copy the result int into strfmt.Datetime.

Reply:
Got it. Done.

Reply:
note the bsonprim.DateTime has no BsonUnmarshal method, only JSON.

Reply:
What I meant is something like this:
```go
i := bsonprim.DateTime{}
i.UnmarshalJSON(data)
```
Then copy i into t. Not sure if this will work for all test cases.

Reply:
I will do a test later but it doesn't seem right to me to use a JSON unmarshaler to unmarshal BSON. Also we will have to do the `tpe == bsontype.Null` check anyway.

Reply:
I tried it. It does not work to use json Unmarshal.
Can you revert to the commit before?

Reply:
for example, this code is completely useles in BSON https://github.com/mongodb/mongo-go-driver/blob/0cc7c8bc77bcb21feae730eb4e6de31609d3545b/bson/primitive/primitive.go#L56. Also that method might change  and break sstrfmt. I suggest we don't unmarshal BSON witht the JSON unmarshaler.

Reply:
Reverted to the previous commit. Can you also create a new tag after merge?